### ISSUE TITLE
Possible to include caps hats, beards for headshots?

### ISSUE THREAD

**Issue by markreg** [wavexx/facedetect#22]
Hello, I'm working on making a headshot program, similar to ID card photographs for my office. Is it possible to optionally include caps, hats and beards included within the boundaries of the detected face?

Thanks

**Comment by wavexx** [comment]
facedetect only aims to return coordinates of the faces. You can use the returned values yourself or modify the source directly. I've also seen a library in PyPy based on this source that you could use.

**Comment by markreg** [comment]
Yes, currently I adjust the coordinates and expand the area of the image by 4 to approximately include all caps, hats, etc.

My main question relates to how the face detection algorithm itself can be tweaked to include the entire person's head, including caps, hats, beards, etc. within the coordinates. Is there another library that can reveal coordinates for a person's head?

**Comment by wavexx** [comment]
On Wed, May 02 2018, markreg wrote:
> My main question relates to how the face detection algorithm itself
> can be tweaked to include the entire person's head, including caps,
> hats, beards, etc. within the coordinates. Is there another library
> that can reveal coordinates for a person's head?

The short anwer is yes. The OpenCV classifier has several pre-trained
models for specific parts, for example eyes, ears, and so on.

Look at the opencv-data package to see which ones are already available.
Those models can be basically used by changing the profile pathname in
the source.

And I think I've seen a project that does that too (facerec if I'm not
mistaken: https://github.com/shhavel/facerec/).

That being said, none of the opencv models are particularly
high-quality.

I've been using some public tensorflow models which much higher
accuracy. The downside is that TF models are also much larger and most
of the dependencies are not installable on ubuntu or debian.

If you don't have this limitation, and have programming experience, I
really encourage you to look into that direction.
